corehq/apps/case_search/utils.py

```diff
@@ -265,22 +273,20 @@ def get_related_cases(helper, app_id, case_types, cases, custom_related_case_pro
         rel for rels in [get_related_case_relationships(app, case_type) for case_type in case_types]
         for rel in rels
     ]
-    child_case_types = [
-        _type for types in [get_child_case_types(app, case_type) for case_type in case_types]
-        for _type in types
-    ]
 
     expanded_case_results = []
     if custom_related_case_property:
         expanded_case_results.extend(get_expanded_case_results(helper, custom_related_case_property, cases))
 
     results = expanded_case_results
     top_level_cases = cases + expanded_case_results
+    top_level_case_ids = {case.case_id for case in top_level_cases}
     if paths:
         results.extend(get_related_case_results(helper, top_level_cases, paths))
 
-    if child_case_types:
-        results.extend(get_child_case_results(helper, top_level_cases, child_case_types))
+    defined_cases = get_defined_cases(helper, app, case_types, top_level_case_ids, include_related_cases)
```

Comment:
what does `defined` mean in this context?

Reply:
There were overlapping meanings of "get related cases" so I renamed them to essentially be:
High Level: `get_related_cases_result` (previously `get_defined_cases`

new feature: `_get_all_related_cases`
else: 1.`_get_search_detail_path_defined_cases` (previously named `get_related_case_results`) and 2. `_get_child_cases_referenced_in_app`

[851645a](https://github.com/dimagi/commcare-hq/pull/32587/commits/851645acd35d3d0ab5762ef780cfceb61fd176d6)
[6fd4217](https://github.com/dimagi/commcare-hq/pull/32587/commits/6fd42170973c08ee8126b8bd5c098e7163c9551a)